torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1530,12 +1593,21 @@ def _verify(expr, potential_expr):
             for i, ss in enumerate(t.size()):
                 property_source = TensorPropertySource(source, TensorProperty.SIZE, i)
                 track_symint(property_source, ss)
-                if _is_dim_dynamic(t, i):
-                    # If this dim is marked dynamic, we need to do a test on it, to ensure that it has not bee
-                    # constrained to an integer.
-                    if _is_int(ss):
-                        raise RuntimeError(f"Attempting to constrain dim {i} for {source}, which violates user's mark_dynamic")
-                    dynamic_sources.append(property_source)
+                if dynamic_ranges:
+                    dyn_range = dynamic_ranges[pos]
+                    if dyn_range and i in dyn_range:
+                        # If this dim is marked dynamic, we need to do a test on it, to ensure that it has not bee
+                        # constrained to an integer.
+                        if _is_int(ss):
+                            raise RuntimeError(f"Attempting to constrain dimension "
+                                               f"{source.name()}.size()[{i}] to {int(ss)}, "
+                                               "which violates user's constraints")
```

Comment:
What user constraint does it violate. The error message will be much better if you can say what constraint is being violated.